community/primitive-collections/src/main/java/org/neo4j/collection/primitive/PrimitiveIntCollections.java

```diff
@@ -826,4 +827,32 @@ public static boolean contains( int[] values, int candidate )
                     " when adding items uniquely to a collection:" + collection );
         }
     }
+
+    /**
+     * Deduplicates values in the sorted {@code values} array.
+     *
+     * @param values sorted array of int values.
+     * @return the provided array if no duplicates were found, otherwise a new shorter array w/o duplicates.
+     */
+    public static int[] dedup( int[] values )
+    {
+        int unique = 0;
+        for ( int i = 0; i < values.length; i++ )
+        {
+            int value = values[i];
+            for ( int j = 0; j < unique; j++ )
```

Comment:
It goes the other way around, doesn't it, from 0 to `unique` which starts at 0 and grows.